pycbc/conversions.py

```diff
@@ -421,8 +423,60 @@ def lambda_from_mass_tov_file(mass, tov_file, distance=0.):
     return lambdav
 
 
+def ensure_obj1_is_primary(mass1, mass2, *params):
+    """
+    Enforce that the object labelled as 1 is the primary.
+
+    Parameters
+    ----------
+    mass1 : float, numpy.array
+        Mass values labelled as 1.
+    mass2 : float, numpy.array
+        Mass values labelled as 2.
+    *params :
+        The binary parameters to be swapped around when mass1 < mass2.
+        The list must have length 2N and it must be organized so that
+        params[i] and params[i+1] are the same kind of quantity, but
+        for object 1 and object 2, respsectively.
+        E.g., spin1z, spin2z, lambda1, lambda2.
+
+    Returns
+    -------
+    list :
+        A list with mass1, mass2, params as arrays, with elements, each
+        with elements re-arranged so that object 1 is the primary.
+    """
+    # Check params are 2N
+    if len(params) % 2 != 0:
+        raise ValueError("params must be 2N floats or arrays")
+    input_properties, input_is_array = ensurearray((mass1, mass2)+params)
+    # Check inputs are all the same length
+    shapes = [par.shape for par in input_properties]
+    if len(set(shapes)) != 1:
+        raise ValueError("Individual masses and params must have same shape")
+    # What needs to be swapped
+    mask = mass1 < mass2
+    # Output containter
+    output_properties = []
+    for i in numpy.arange(0, len(shapes), 2):
+        # primary (p)
+        p = copy.copy(input_properties[i])
+        # secondary (s)
+        s = copy.copy(input_properties[i+1])
+        # Swap
+        p[mask] = input_properties[i+1][mask]
+        s[mask] = input_properties[i][mask]
+        # Format and include in output object
+        output_properties.append(formatreturn(p, input_is_array))
+        output_properties.append(formatreturn(s, input_is_array))
+    # Release output
+    return output_properties
+
+
 def remnant_mass_from_mass1_mass2_spherical_spin_eos(
-        mass1, mass2, spin1a=0.0, spin1pol=0.0, eos='2H'):
+        mass1, mass2, spin1a=0.0, spin1pol=0.0, eos='2H',
+        spin2a=0.0, spin2pol=0.0, swap_companions=False,
+        max_ns_mass=None, extrapolate=False):
```

Comment:
This is super useful! I had a quick look through (as I was considering using an em-bright cut in a new early warning bank).

There is one thing I would suggest: The `max_ns_mass` should be determined by the function you give it as well, i.e. `max_ns_mass` should be the minimum of the supplied value and the maximum supported value in the equation

Reply:
Thanks, @GarethCabournDavies!
This would require reading the EOS data twice: once to create the `mask` with mass2 <= min(max_ns_mass_given_by_the_user, max_ns_mass_given_by_the_eos), and then a second time when the EOS data is needed to calculate the baryonic mass and the compactness.  This is not ideal, but the rise in computational cost of this can be avoided by, say, maintaining a dictionary of maximum masses.
The thing that concerns me more is the case of a user that does not read the warnings I would send out instead of raising a `ValueError` (as is the case right now) when min(max_ns_mass_given_by_the_user, max_ns_mass_given_by_the_eos) = max_ns_mass_given_by_the_eos.  That user could, for example, extract N values of NS mass between 1 and 10 solar masses, calculate the remnant mass with all other parameters being fix, and conclude that a high (wrong!) fraction of NSBHs does not produce a remnant mass.  This is pretty extreme, of course, but if you get around ~3 solar masses things get dangerous.  The first time one extracts NS masses between 1 and 3 Msun, with the 2H EOS, and performs remnant mass calculations, they should learn from a code error that this EOS does not go above >~ 2.8 Msun, and set `max_ns_mass`.  Maybe this needs a better name, e.g., `ns_bh_mass_boundar`.  I just don't want the code to handle NSBHs as BBHs without the user knowing.